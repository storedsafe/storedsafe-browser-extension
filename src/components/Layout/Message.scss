@import '../../variables';

.message {
  padding: $spacing;
  border: 0 solid $color-accent-0;
  border-left-width: 6px;
  background-color: $color-accent-1;
  color: $color-accent-3;
  border-radius: $border-radius;

  &.warning {
    border-color: $color-warning-0;
    background-color: $color-warning-1;
    color: $color-warning-3;
  }

  &.error {
    border-color: $color-error-0;
    background-color: $color-error-1;
    color: $color-error-3;
  }
}
