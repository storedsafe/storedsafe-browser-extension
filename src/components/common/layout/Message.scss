@import '../../../styles/variables.scss';

.message {
  padding: $spacing;
  border: 0 solid $color-accent-0;
  border-left-width: 6px;
  background-color: $color-accent-1;
  color: $color-accent-3;

  &.warning {
    border-color: $color-warning-0;
    background-color: $color-warning-1;
    color: $color-warning-3;
  }

  &.error {
    border-color: $color-danger-0;
    background-color: $color-danger-1;
    color: $color-danger-3;
  }
}
