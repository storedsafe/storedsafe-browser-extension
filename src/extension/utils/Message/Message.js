const MessageType = Object.freeze({
  LOGIN: 'login',
  LOGOUT: 'logout',
  GET_SETTINGS: 'getSettings',
  GET_TOKEN: 'getToken',
});

function sendMessage(messageType, data) {
  return browser.runtime.sendMessage({ messageType, data });
}

export { MessageType };
export default Object.freeze({
  login: (data) => (sendMessage(MessageType.LOGIN, data)),
  logout: (data) => (sendMessage(MessageType.LOGOUT, data)),
  getSettings: (data) => (sendMessage(MessageType.GET_SETTINGS, data)),
  updateSettings: (data) => (sendMessage(MessageType.UPDATE_SETTINGS, data)),
  getToken: (data) => (sendMessage(MessageType.GET_TOKEN, data)),
});
