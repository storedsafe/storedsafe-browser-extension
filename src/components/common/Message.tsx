import React from 'react'
import './Message.scss'

interface MessageProps {
  type?: 'info' | 'warning' | 'error'
  children?: React.ReactNode
}

export const Message: React.FunctionComponent<MessageProps> = ({
  type,
  children
}: MessageProps) => {
  return (
    <div className={`message ${type}`}>
      {children}
    </div>
  )
}

Message.defaultProps = {
  type: 'info'
}
