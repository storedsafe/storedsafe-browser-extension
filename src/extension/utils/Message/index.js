import Message, { MessageType } from './Message';

export { MessageType };
export default Message;
