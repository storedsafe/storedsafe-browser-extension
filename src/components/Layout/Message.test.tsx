import * as React from 'react';
import { shallow } from 'enzyme';
import { Message } from './Message';

test('<Message />', () => {
  const wrapper = shallow(<Message />);
  expect(wrapper).toMatchSnapshot();
});

test('<Message type=info />', () => {
  const wrapper = shallow(<Message type="info" />);
  expect(wrapper).toMatchSnapshot();
});

test('<Message type=warning />', () => {
  const wrapper = shallow(<Message type="warning" />);
  expect(wrapper).toMatchSnapshot();
});

test('<Message type=error />', () => {
  const wrapper = shallow(<Message type="error" />);
  expect(wrapper).toMatchSnapshot();
});

test('<Message />, with children', () => {
  const wrapper = shallow(<Message>children</Message>);
  expect(wrapper.text()).toBe('children');
  expect(wrapper).toMatchSnapshot();
});

