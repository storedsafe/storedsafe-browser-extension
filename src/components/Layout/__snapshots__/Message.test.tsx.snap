// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> 1`] = `ShallowWrapper {}`;

exports[`<Message />, with children 1`] = `ShallowWrapper {}`;

exports[`<Message type=error /> 1`] = `ShallowWrapper {}`;

exports[`<Message type=info /> 1`] = `ShallowWrapper {}`;

exports[`<Message type=warning /> 1`] = `ShallowWrapper {}`;
